### ISSUE TITLE
Guests entering an over-capacity queue and "lining up twice"

### ISSUE THREAD

**Issue by CDB-Man** [OpenRCT2/OpenRCT2#17814]
### Operating System

Windows 10

### OpenRCT2 build

OpenRCT2, v0.4.1-185-g53f7a20 (53f7a20 on develop) provided by GitHub

### Describe the issue

Somehow, this queue line goes over-capacity.  See screenshot, based on this queue length, I would expect ~60 guests, but you can see the queue has 159 people per the Cape Canaveral ride window, and visually, there are a lot of overlapping guests stacked on top of each other in the overworld.  See how at the front of the line, there's no spacing at all between guests because they are overlapping on top of each other, whereas in the pack the spacing is more of what is expected.

I have noticed that even though the line is full, guests are still entering the line, and that they are going all the way to the front, as if ignoring the existing queued guests entirely.

The impact of this is that the queue will exponentially grow, leading to very long queue times longer than the ride capacity, leading to unhappy guests that are waiting in line too long since the queue is over designed capacity.

![image](https://user-images.githubusercontent.com/14037303/184517037-d8e673f0-6db6-4af0-b29c-0e9a81935f24.png)

### Steps to reproduce

1. Load the attached Ivory Towers save file
2. Go to Cape Canaveral, and you will see that guests enter the queue even though the queue is full, and they walk directly to the front of the queue

### Attachments

[Ivory Towers.zip](https://github.com/OpenRCT2/OpenRCT2/files/9332435/Ivory.Towers.zip)

**Comment by Kuhnovic** [comment]
I've played around with your save file and I think I've narrowed down the problem a bit. It seem that if you change the layout of the queue line path for a particular ride, any peeps currently walking towards that ride will start cutting the line upon arrival. Either they walk all the way to the front of the line, or they "double queue". Depending on the size of the park and the amount of peeps walking towards the ride it can take a long time for these effects to dissipate.

Here you see me reproduce it with a Merry-go-round in Forest Frontiers:

![uSahwynhlI](https://user-images.githubusercontent.com/68320206/184960890-76cfd9ab-3d6b-4eab-b6e5-ef9b19c942fe.gif)
.

You probably made a path change and then saw these strange effects. All other rides in your park looked normal. Since your park was reasonably big it probably took a long time for all peeps heading for the Cape Carnaveral to actually get there, and even for longer for the double queueing to disappear.
